app/src/main/java/org/fossasia/openevent/fragments/SpeakerFragment.java

```diff
@@ -25,19 +36,34 @@
  * create an instance of this fragment.
  */
 public class SpeakerFragment extends Fragment {
+    private SwipeRefreshLayout swipeRefreshLayout;
 
-    RecyclerView speakersRecyclerView;
-    SpeakersListAdapter speakersListAdapter;
-    DbSingleton dbSingleton = DbSingleton.getInstance();
+    private RecyclerView speakersRecyclerView;
+    private SpeakersListAdapter speakersListAdapter;
 
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        Bus bus = OpenEventApp.getEventBus();
+        bus.register(this);
```

Comment:
where is the unregister getting done? The bus needs to be unregistered as well, you should do it according to fragment lifecycle, otherwise it will be a memory leak, the fragment will forever listen to events.